### PR TITLE
Add Product and Search breadcrumb

### PR DESCRIPTION
**Pull request by klzns**
**What problem is this solving?**

Render the breadcrumb structured data.

Breadcrumb docs reference: https://developers.google.com/search/docs/data-types/breadcrumb

**How should this be manually tested?**

Product:
https://breadcrumb--storecomponents.myvtex.com/wood-clock/p
image

Search:
https://breadcrumb--storecomponents.myvtex.com/apparel---accessories/hats
image

---

Related to:
- https://github.com/vtex-apps/breadcrumb/pull/70
- https://github.com/vtex-apps/store-discussion/issues/318

### CODE REVIEW

File: react/ProductBreadcrumb.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            item: `${baseUrl}/${productSlug}/p`,
+          },
+        ]
+      : []
+
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'BreadcrumbList',
+    itemListElement: categoryItems.concat(...productItem),
+  }
```
Comment (kaisermann) [inline react/ProductBreadcrumb.tsx:45]: what do you think of:

```suggestion
if (productName && productSlug) {
  categoryItems.push({
    '@type': 'ListItem',
    position: categoryItems.length + 1,
    name: productName,
    item: `${baseUrl}/${productSlug}/p`,
  })
}

return {
  '@context': 'https://schema.org',
  '@type': 'BreadcrumbList',
  itemListElement: categoryItems,
}
```

File: react/ProductBreadcrumb.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  productName?: string
+  productSlug?: string
+}
+
+const ProductBreadcrumbStructuredData: FC<Props> = ({
+  categoryTree,
+  productName,
+  productSlug,
+}) => {
+  const breadcrumbLD = productBreadcrumb(categoryTree, productName, productSlug)
```
Comment (kaisermann) [inline react/ProductBreadcrumb.tsx:59]: `getProductBreadcrumb` 👀 ?

File: react/SearchBreadcrumb.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import { BreadcrumbList } from './modules/breadcrumb'
+import { getBaseUrl } from './modules/baseUrl'
+
+interface SearchBreadcrumbItem {
+  name: string
+  href: string
+}
+
+const searchBreadcrumb = (
```
Comment (kaisermann) [inline react/SearchBreadcrumb.tsx:11]: `getSearchBreadcrumb` 👀 ?

File: react/SearchBreadcrumb.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  breadcrumb?: SearchBreadcrumbItem[]
+): BreadcrumbList | {} => {
+  if (!breadcrumb || !Array.isArray(breadcrumb)) {
```
Comment (kaisermann) [inline react/SearchBreadcrumb.tsx:14]: I think the `isArray` check is enough here 

```suggestion
  if (!Array.isArray(breadcrumb)) {
```

File: react/ProductBreadcrumb.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  name: string
+  href: string
+}
+
+export const productBreadcrumb = (
+  categoryTree?: CategoryTreeItem[],
+  productName?: string,
+  productSlug?: string
+): BreadcrumbList | {} => {
+  if (!categoryTree || !Array.isArray(categoryTree)) {
```
Comment (kaisermann) [inline react/ProductBreadcrumb.tsx:16]:
```suggestion
  if (!Array.isArray(categoryTree)) {
```

File: react/ProductBreadcrumb.tsx (hunk 1 above)
Comment (klzns) [reply react/ProductBreadcrumb.tsx:45]: uhh much better 👌

**Pull request merged**